### PR TITLE
add beta4 changelog entries

### PR DESCRIPTION
**Pull request by lboeman**
Adds changelog documentation for beta 4. This may be incomplete and should probably stay open until just before tagging.

### CODE REVIEW

**Comment by wholmgren** [comment]
xref https://github.com/SolarArbiter/SolarArbiter.github.io/pull/125

**Comment by alorenzo175** [comment]
seems like this is going on the github page, so I'm cutting beta4 now

**Comment by wholmgren** [comment]
The link still points to https://github.com/SolarArbiter/solarforecastarbiter_dashboard/blob/master/CHANGELOG.md so we'll need to merge this when it's ready.

@alorenzo175 any items you want to add?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,40 @@ in this file.
...
+
+- Reports now contain a table of all included metrics over the entire selected
+  period.
+
+- Reports in the *pending* and *failed* state now display a message to the user
+  about their status. For failed reports, this is a list of errors encountered
+  while processing the report.
+
+- The dashboard report view now allows users to search metric plots by
+  forecast ant to sort metric plots by metric, category and forecast.
```
Comment (wholmgren) [inline CHANGELOG.md]:
```suggestion
  forecast and to sort metric plots by metric, category and forecast.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,40 @@ in this file.
...
+
+- Dashboard users can now download report metrics as a csv.
+
```
Comment (wholmgren) [inline CHANGELOG.md]:
```suggestion
- Daily updating precomputed reports for select reference data.
```

**Comment by wholmgren** [comment]
@lboeman can you reorder the `Added` list to be most to least interesting? Did you also change the dashboard download behavior for this release? You might double check the closed PRs...

I reviewed the core repository change log and I think all of the enhancements are accounted for here. Should we include these bug fixes here too? They would need some rewording to make sense to the dashboard user.

```
Fix handling of empty observation timeseries in metrics preprocessing. (GH295) (GH296)
Fix handling of interval_label == ending in the groupby categories in metrics.calculator. (GH234) (GH297)
```

**Comment by lboeman** [comment]
> @lboeman can you reorder the `Added` list to be most to least interesting? Did you also change the dashboard download behavior for this release? You might double check the closed PRs...
 ```
I've reordered the *Added* list with the idea that new *thing* for users to look at go at the top (e.g. the reference reports), then new functionality, and then UI updates. 

I added those two core bug fixes. They seem relevant given the point about failed reports.

**Pull request merged**